SwiftUIProject/SwiftUIProject/Products/ProductsView.swift

```diff
@@ -0,0 +1,215 @@
+                    .resizable()
+                    .foregroundColor(.gray.opacity(0.3))
+                    .frame(
+                        width: 20,
+                        height: 20
+                    )
+                    .fontWeight(.semibold)
+                Text(Constants.searchBarText)
+                    .foregroundColor(viewModel.customPurpleColor)
+                Spacer()
+            }
+            .padding()
+        }
+        .frame(
+            width: 300,
+            height: 50,
+            alignment: .leading
+        )
+        .padding(.vertical)
+    }
+
+    private var totalPriceView: some View {
+        VStack {
+            Text("\(Constants.totalPriceViewText)\(viewModel.totalPrice)")
+                .font(.system(
+                    size: 22,
+                    weight: .bold
+                ))
+                .padding()
+            Text("\(Constants.amountProductsText)\(viewModel.amountFurniture)")
+        }
+    }
+
+    private var purchaseButtonView: some View {
+        NavigationLink {
+            FiltersView()
+        } label: {
+            Image(systemName: Constants.cartImageName)
+                .resizable()
+                .foregroundColor(.orange)
+                .frame(
+                    width: 40,
+                    height: 40
+                )
+        }
+    }
+
+    private var productRectangle: some View {
+        Rectangle()
+            .frame(
+                width: 300,
+                height: 100
+            )
+            .foregroundColor(.white)
+            .shadow(
+                color: .black.opacity(0.2),
+                radius: 5
+            )
+    }
+
+    // MARK: - Private Methods
+
+    private func makeOneProductView(
+        product: Product,
+        imageWidth: CGFloat
+    ) -> some View {
+        ZStack {
+            productRectangle
+            HStack {
+                makeImageView(imageName: product.imageName)
+                Spacer()
+                makeNameAndAmountView(product: product)
+                Spacer()
+                makePriceView(price: product.price)
+            }
+            .padding(.horizontal, 60)
+        }
+    }
+
+    private func makeImageView(imageName: String) -> some View {
+        Image(systemName: imageName)
+            .resizable()
+            .foregroundColor(.orange)
+            .fontWeight(.light)
+            .scaledToFit()
+            .frame(
+                width: 50,
+                height: 50
+            )
+    }
+
+    private func makePriceView(price: Int) -> some View {
+        VStack {
+            Text(Constants.saleText)
+                .foregroundColor(.gray.opacity(0.2))
+                .font(.system(
+                    size: 20,
+                    weight: .bold
+                ))
+                .strikethrough(
+                    true,
+                    color: .gray.opacity(0.2)
+                )
+            Text("$\(price)")
```

Comment:
```suggestion
            Text("$\(price)")
```
а вот доллар в константы)